libs/ngx-nuts-and-bolts/src/lib/utilities/table-state/table-state.spec.ts

```diff
@@ -0,0 +1,195 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+/* eslint-disable */
+/* eslint-disable */
+import { TestBed } from '@angular/core/testing';
+import { ActivatedRoute, Router } from '@angular/router';
+import { RouterTestingModule } from '@angular/router/testing';
+import { take } from 'rxjs/operators';
+import {
+	changeFilters,
+	changePage,
+	changeSort,
+	createCustomFiltersObservable,
+	createPaginationObservable,
+	createSortObservable,
+	IPageInfo,
+	ISortInfo,
+} from './table-state';
+
+describe('Table sate', () => {
+	let navigationSpy: jest.SpyInstance<Promise<boolean>>;
+	let router: Router;
+	let route: ActivatedRoute;
+
+	beforeEach(async () => {
+		await TestBed.configureTestingModule({
+			imports: [RouterTestingModule],
+		});
+
+		router = TestBed.inject(Router);
+		route = TestBed.inject(ActivatedRoute);
+
+		navigationSpy = jest.spyOn(router, 'navigate');
+	});
+
+	it('should set query parameters for pagination', async () => {
+		expect(navigationSpy).toHaveBeenCalledTimes(0);
+		let paginationValue = await createPaginationObservable(route).pipe(take(1)).toPromise();
```

Comment:
I think this test is testing more than necessary. You only want to test if the query parameters are set. You should not care about the pagination observable, that observable can be tested separately.

I think that all you need in this test is this:
```ts
expect(navigationSpy).toHaveBeenCalledTimes(0);

const pageInfo: IPageInfo = {
  pageIndex: 1,
  pageSize: 12,
};

await changePage(router, pageInfo);
expect(navigationSpy).toHaveBeenCalledTimes(1);
expect(navigationSpy).toHaveBeenCalledWith(...);
```

You must check that the navigate spy was called with the correct options that will change the query params. This is an important part that you are missing.